src/mpid/ch3/src/ch3u_win_fns.c

```diff
@@ -356,6 +356,18 @@ int MPID_Win_set_info(MPIR_Win * win, MPIR_Info * info)
             if (!strncmp(info_value, "false", sizeof("false")))
                 win->info_args.same_disp_unit = FALSE;
         }
+
+        /********************************************************/
+        /******* accept (but ignore) a CH4-specific hint ********/
+        /********************************************************/
+        info_flag = 0;
+        MPIR_Info_get_impl(info, "optimized_mr", MPI_MAX_INFO_VAL, info_value, &info_flag);
+        if (info_flag) {
+            if (!strncmp(info_value, "true", sizeof("true")))
+                win->info_args.optimized_mr = TRUE;
+            if (!strncmp(info_value, "false", sizeof("false")))
+                win->info_args.optimized_mr = FALSE;
```

Comment:
It doesn't necessarily mean not supported. It just indicates what the user asked for. The issue would be that if we always set it to `FALSE`, the currently failing test will still fail.

Reply:
Not all libfabric providers support optimized_mr, right? How does current code and the test deal with that?

Reply:
Again. The only thing the info hint is indicating is that the user wants to use optimized MRs and has fulfilled all of the requirements (if any exist). That can mean different things for different devices/netmods/providers. For CH3, it means MPICH is aware that the user has asked for it and is just returning that value. For CH4/UCX, it means the same thing. For the CH4/OFI with the cxi provider, it has a specific meaning and requirement.

Reply:
I see. Then I question this implementation. The only reason user call `MPI_Win_info_get` is to get feedback from implementation whether certain hints is accepted. By accepted, it usually means it will do what the user is requesting. If we simply echo what user asked back, then there is zero information we are returning.  Actually, it is worse since we are giving user a fake or even wrong feedback.

Reply:
I think maybe the way this discussion should go is whether or not the changes in [`6f25672` (#5710)](https://github.com/pmodels/mpich/pull/5710/commits/6f25672cfd3f34945848493bf0a4274abe2e1588) are valid. Maybe that test should just be dropped since I'm not sure if it will actually convey any information anyway.

Reply:
Yeah, the discussion applies to what we did for non-CXI providers and ucx as well.

Reply:
So should we just revert that patch and close this PR?

Reply:
Let's discuss and reach a conclusion first :) My suggestion is for providers or ch3 that doesn't support `optimized_mr`, we just return the infokey in `MPI_Win_get_info` with value set to `false`. For CXI provider, we certainly should return true or false based on whether it is actually enabled.

Reply:
I think we should just ignore this hint for everything that doesn't do something with it.

Reply:
I see. So the question is just how to test it in a portable way?

Reply:
The main issue (and I assume the reason this wasn't implemented this way in the first place) is that OFI passes all hint handling on to the `MPIDIG` code, which would mean adding a bunch of extra code in `ofi_win.c` to handle info hints would be a pain and I'm not sure how much value it would provide over what it's doing right now.

There's not requirements in the MPI Standard about what the info hints return. There was recently a question about whether or not an implementation could return something other than what the user provided (e.g., could the user ask to align bytes at 21 bytes and the implementation change it to 16, or could the user ask to turn on `optimized_mr` and the implementation say no). I don't remember what the specific answer here was, but I think the intention was for the implementation to be *allowed*, but not *required* to convey information about what it's actually doing, rather than what the user specifically asked for.

Reply:
I see. Let me think about it. Maybe I will open an PR to show what I have in mind and then we can move the discussion over there.

Reply:
Ideally we'd do things in the same way we do for communicators, where we allow pretty much every level to manage its own hints. That way we can throw extra logic in OFI to detect which provider we're using and whether or not it has the "optimized MR" capability checked. If so, we'll accept the hint. Otherwise, we'll ignore it or set it to `false`. If you're not running ch4/ofi, you'd just ignore the hint completely.

Reply:
I agree with this, and I think we should do that rather than rendering `MPI_Win_get_info` useless.